robot/impl/resource_manager.go

```diff
@@ -472,6 +472,14 @@ func (manager *resourceManager) completeConfig(
 		if !ok || !gNode.NeedsReconfigure() {
 			continue
 		}
+		if !gNode.ShouldTryReconfigure() {
+			err := errors.Errorf("Reconfiguration error: reached max reconfiguration attempts for %s", gNode.Config().Name)
```

Comment:
I don't _love_ the messaging of this error just because this might be happening because of an inability to _build_ initially and not _reconfigure_. Let me think about alternatives 🤔 .

Reply:
Based on if a node.IsUninitialized(), we could change the error message?

Reply:
Yep that's a good idea; that should work.